src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetCoreMvc2Integration.cs

```diff
@@ -250,6 +256,79 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
             }
         }
 
+        /// <summary>
+        /// Wrapper method used to catch unhandled exceptions in the incoming request pipeline for Microsoft.AspNetCore.Mvc.Core
+        /// </summary>
+        /// <param name="context">The DiagnosticSource that this extension method was called on.</param>
+        [InterceptMethod(
+            CallerAssembly = "Microsoft.AspNetCore.Mvc.Core",
+            TargetAssembly = "Microsoft.AspNetCore.Mvc.Core",
+            TargetType = ExceptionHookType)]
+        public static void Rethrow(object context)
```

Comment:
This is actually a set of private methods within Microsoft.AspNetCore.Mvc.Core
I would like to use a public method to access any unhandled exceptions, but I ran into some problems.
The exception itself is swallowed before any of the public methods exit, and would require grabbing a reference to the private properties within some of the ResourceInvoker class, which ends up being a worse option than instrumenting the private methods in my opinion. Another *potential* option is picking up the exception later in the pipeline, which might make the trace look odd?

Another problem I encountered is that even though there was an unhandled exception in the pipeline, the diagnostics method we use doesn't yet have the http status code updated. That will need to be visited.

Reply:
> The exception itself is swallowed before any of the public methods exit, and would require grabbing a reference to the private properties within some of the ResourceInvoker class, which ends up being a worse option than instrumenting the private methods in my opinion. 

I don't know what is the drawbacks between the two options. Can you argument more?

> Another potential option is picking up the exception later in the pipeline, which might make the trace look odd?

In some cases (i.e. Rails), we collect exceptions later in the pipeline but we update the (unfinished) `Span` the exception was related. It means that maybe we have already other spans in-flight, but still capable to set that value to one of the parents (the one that makes sense of course). If you don't have a similar case (i.e. is _too_ late or you'll end up attaching the exception in a wrong place), then I would say to ignore this option.

In the overall, what's the risk of instrumenting this private method? Can we crash if methods are renamed, or we simply stop collecting the exception (so it's just a regression?).

Reply:
Accessing a private property would mean making changes to the c++ 
The properties are far more likely to change and have contention as well.

I was looking for a way to pick up the exception later in the pipeline and hadn't found one, but I will take another look.

If the method is renamed, then we would simply not instrument it (a regression).

Reply:
Also, the private methods in this signature have been there since October 2016, and with additions in 2017, but no changes to the signature. So this should work for versions all the way back to 1.0.0 based on dates in commit history.